waspc/data/Generator/templates/react-app/src/queries/index.js

```diff
@@ -1,18 +1,16 @@
 import { useQuery as rqUseQuery } from 'react-query'
 
-export const useQuery = (queryFn, queryFnArgs, config) => {
+export function useQuery(queryFn, queryFnArgs, config) {
```

Comment:
We might want to reconsider whether this API should look like `react-query`'s:
```javascript
 useQuery({
   queryKey: 'posts',
   queryFn: fetchPosts,
 });
```

Not only to make our interface as close as possible to `react-query`'s, but also because sending arguments as object props is more flexible (i.e., we can add new arguments in a backwards compatible way, we don't need to pass in the dummy data if we want to omit something) and arguably more clear (i.e., arguments have labels, no need to memorize the ordering) than using positional arguments.

Reply:
I don't know all of the intricacies here, but that general approach sounds logical to me. 👍🏻

Reply:
Hm, on second thought, I think we should offer both approaches as well.

It doesn't cost us anything and is pretty flexible. I personally prefer the object approach, but a lot of people might not (especially because some IDEs add argument name labels to function calls).

@Martinsos I'll make an issue to add support for object arguments.

Reply:
Ok sounds good, let's do both then! That resolves the other discussions also, since we don't have to choose.